### PR TITLE
improve file path validation when reading parquet

### PR DESCRIPTION
**Pull request by Weijun-H**
## Which issue does this PR close?

Follow #8265

### CODE REVIEW

File: datafusion/core/src/execution/context/mod.rs [hunk 1]
```diff
@@ -858,10 +858,14 @@ impl SessionContext {
 
         // check if the file extension matches the expected extension
         for path in &table_paths {
-            let file_name = path.prefix().filename().unwrap_or_default();
-            if !path.as_str().ends_with(&option_extension) && file_name.contains('.') {
+            let file_path = path.as_str();
+            if (!file_path.ends_with(option_extension.clone().as_str())
+                && !path.is_collection())
+                || (!file_path.ends_with((option_extension.clone() + "/").as_str())
```
[the remark this answers is not in the document]
Comment (andygrove) [reply datafusion/core/src/execution/context/mod.rs]: You could use `std::path::MAIN_SEPARATOR` instead here

**Review by comphead — COMMENTED** [review]
Thanks @Weijun-H for a quick turnaround. Please add a test for double dots

```
bbb..bbb/filename.parquet
```

File: datafusion/core/src/execution/context/parquet.rs [hunk 2]
```diff
@@ -138,10 +138,10 @@ mod tests {
         Ok(())
     }
 
-    #[cfg(not(target_family = "windows"))]
```
Comment (Weijun-H) [inline datafusion/core/src/execution/context/parquet.rs:141]: extend to all platform

**Review by andygrove — APPROVED** [review]
Thanks @Weijun-H

**Review by comphead — APPROVED** [review]
lgtm thanks @Weijun-H

**Pull request merged**